pkg/asset/installconfig/platform.go

```diff
@@ -57,61 +61,80 @@ func (a *Platform) Name() string {
 	return "Platform"
 }
 
-func (a *Platform) queryUserForPlatform() string {
-	var platform string
-	survey.AskOne(&survey.Select{
-		Message: "Platform",
-		Options: validPlatforms,
-	}, &platform, nil)
+func (a *Platform) queryUserForPlatform() (string, error) {
+	prompt := asset.UserProvided{
+		Prompt: &survey.Select{
+			Message: "Platform",
+			Options: validPlatforms,
+		},
+		EnvVarName: "OPENSHIFT_INSTALL_PLATFORM",
+	}
+
+	platform, err := prompt.Generate(nil)
+	if err != nil {
+		return "", err
+	}
 
-	return platform
+	return string(platform.Contents[0].Data), nil
 }
 
 func (a *Platform) awsPlatform() (*asset.State, error) {
-	var region string
-	survey.AskOne(&survey.Select{
-		Message: "Region",
-		Help:    "The AWS region to be used for installation.",
-		Default: "us-east-1 (N. Virginia)",
-		Options: []string{
-			"us-east-2 (Ohio)",
-			"us-east-1 (N. Virginia)",
-			"us-west-1 (N. California)",
-			"us-west-2 (Oregon)",
-			"ap-south-1 (Mumbai)",
-			"ap-northeast-2 (Seoul)",
-			"ap-northeast-3 (Osaka-Local)",
-			"ap-southeast-1 (Singapore)",
-			"ap-southeast-2 (Sydney)",
-			"ap-northeast-1 (Tokyo)",
-			"ca-central-1 (Central)",
-			"cn-north-1 (Beijing)",
-			"cn-northwest-1 (Ningxia)",
-			"eu-central-1 (Frankfurt)",
-			"eu-west-1 (Ireland)",
-			"eu-west-2 (London)",
-			"eu-west-3 (Paris)",
-			"sa-east-1 (São Paulo)",
+	prompt := asset.UserProvided{
+		Prompt: &survey.Select{
+			Message: "Region",
+			Help:    "The AWS region to be used for installation.",
+			Default: "us-east-1 (N. Virginia)",
+			Options: []string{
+				"us-east-2 (Ohio)",
+				"us-east-1 (N. Virginia)",
+				"us-west-1 (N. California)",
+				"us-west-2 (Oregon)",
+				"ap-south-1 (Mumbai)",
+				"ap-northeast-2 (Seoul)",
+				"ap-northeast-3 (Osaka-Local)",
+				"ap-southeast-1 (Singapore)",
+				"ap-southeast-2 (Sydney)",
+				"ap-northeast-1 (Tokyo)",
+				"ca-central-1 (Central)",
+				"cn-north-1 (Beijing)",
+				"cn-northwest-1 (Ningxia)",
+				"eu-central-1 (Frankfurt)",
+				"eu-west-1 (Ireland)",
+				"eu-west-2 (London)",
+				"eu-west-3 (Paris)",
+				"sa-east-1 (São Paulo)",
+			},
 		},
-	}, &region, nil)
+		EnvVarName: "OPENSHIFT_INSTALL_AWS_REGION",
+	}
+	region, err := prompt.Generate(nil)
+	if err != nil {
+		return nil, err
+	}
 
 	return assetStateForStringContents(
 		AWSPlatformType,
-		strings.Split(region, " ")[0],
+		strings.Split(string(region.Contents[0].Data), " ")[0],
 	), nil
 }
 
 func (a *Platform) libvirtPlatform() (*asset.State, error) {
-	var uri string
-	survey.AskOne(&survey.Input{
-		Message: "URI",
-		Help:    "The libvirt connection URI to be used. This must be accessible from the running cluster.",
-		Default: "qemu+tcp://192.168.122.1/system",
-	}, &uri, nil)
+	prompt := asset.UserProvided{
+		Prompt: &survey.Input{
```

Comment:
Did you forget to use `&survey.Select` here?

Reply:
Do I understand correctly that due to https://github.com/openshift/installer/pull/320/files#diff-042a71691f31141034ea54bae5b36b06R26 this is settable via environment variable and will only ask interactively if it's not present?

Reply:
Yes.